optimum/habana/transformers/models/glm4v/tokenization_chatglm.py

```diff
@@ -0,0 +1,455 @@
+# coding=utf-8
+# Copyright 2024 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+###############################################################################
+# Copyright (C) 2022-2024 Habana Labs, Ltd. an Intel Company
+###############################################################################
+
+"""
+Adapted from the following source:
+https://huggingface.co/THUDM/glm-4v-9b/blob/main/tokenization_chatglm.py
+"""
+
+import base64
+import json
+import os
+from typing import Any, Dict, List, Optional, Union
+
+import regex as re
+import torch
+from torch import TensorType
+from torchvision import transforms
+from transformers import PreTrainedTokenizer
+from transformers.tokenization_utils_base import BatchEncoding, EncodedInput
+from transformers.utils import PaddingStrategy
+from transformers.utils.import_utils import is_tiktoken_available
+
+
+if is_tiktoken_available():
+    import tiktoken
+else:
+    raise ModuleNotFoundError(
+        "GLM-4V requires the Tiktoken library to be installed. Please install it with: `pip install tiktoken`"
+    )
```

Comment:
```suggestion
```

Reply:
I followed the suggestion as https://github.com/huggingface/optimum-habana/pull/1547/files, and I don't think it should be removed.

Reply:
@mengker33 The error cannot be raised here as it will raise an error if tiktoken is not available for all scripts that call `adapt_transformers_to_gaudi`. Since this method imports classes from this file, the error will be raised. For instance you can try
```
pytest tests/test_text_generation_example.py -v -s -k "llama"
```
and it will fail if tiktoken is not installed, which shouldn't be the case.

That's why I put it at the beginning of the `__init__` for Baichuan. That way I'm sure it is only raised if this model is used, I suggest you do the same here.